composer/algorithms/swa/swa.py

```diff
@@ -96,22 +161,25 @@ def apply(self, event: Event, state: State, logger: Logger) -> None:
                 state.optimizers[0],
                 swa_lr=self.swa_lr,
                 anneal_epochs=self.anneal_epochs,
-                anneal_strategy='cos',
+                anneal_strategy=self.anneal_strategy,
             )
 
         if self.swa_model is None:
             self.swa_model = AveragedModel(state.model)
 
         self.swa_model.update_parameters(state.model)  # type: ignore
 
-        if self.swa_scheduler is None:
-            raise ValueError('SWA LR scheduler was not set.')
-        self.swa_scheduler.step()
-
-        ## end of training
-        if float(state.get_elapsed_duration()) >= 1.0:
-            device = next(self.swa_model.parameters()).device
-            # TODO(laura) this does not apply the batch split fn. This may result in cuda OOM
-            update_bn(state.train_dataloader, model=self.swa_model, device=device)
```

Comment:
I think we still need to have a call to update_bn if we're doing SWA all the way until the end, no? @growlix

Reply:
`update_bn()` does a full epoch of forward passes on the data internally by iterating through `train_dataloader`, which—I think—makes it basically incompatible with Composer because it can't account for microbatching and all the other logic the trainer does in between a single iteration through `train_dataloader` and actually doing a forward pass. @jbloxham and @abhi-mosaic would probably have a better idea than me though

Reply:
For now maybe we should just spit a warning if they set `swa_end="1dur"` or something similar